exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/TraceExporter.java

```diff
@@ -15,134 +15,108 @@
+   * does not throw any exception, an exception may still be thrown during the attempt to generate
+   * the actual {@link TraceExporter}.
+   *
+   * @return A configured instance of {@link TraceExporter} which gets initialized lazily once
+   *     {@link TraceExporter#export(Collection)} is called.
+   */
+  public static SpanExporter createWithDefaultConfiguration() {
+    return generateSkeletonTraceExporter(DEFAULT_CONFIGURATION);
   }
 
-  TraceExporter(
-      String projectId,
-      CloudTraceClient cloudTraceClient,
-      ImmutableMap<String, String> attributeMappings,
-      Map<String, AttributeValue> fixedAttributes) {
-    this.projectId = projectId;
-    this.cloudTraceClient = cloudTraceClient;
-    this.projectName = ProjectName.of(projectId);
-    this.translator = new TraceTranslator(attributeMappings, fixedAttributes);
+  /**
+   * Method that generates an instance of {@link TraceExporter} using a {@link TraceConfiguration}
+   * that allows the user to provide custom configuration for Traces.
+   *
+   * <p>This method defers the creation of an actual {@link TraceExporter} to a point when it is
+   * actually needed - which is when the spans need to be exported. As a result, while this method
+   * does not throw any exception, an exception may still be thrown during the attempt to generate
+   * the actual {@link TraceExporter}.
+   *
+   * @param configuration The {@link TraceConfiguration} object that determines the user preferences
+   *     for Trace.
+   * @return An instance of {@link TraceExporter} as a {@link SpanExporter} object
+   */
+  public static SpanExporter createWithConfiguration(TraceConfiguration configuration) {
+    return generateSkeletonTraceExporter(configuration);
   }
 
   @Override
   public CompletableResultCode flush() {
-    // We do no exporter buffering of spans, so we're always flushed.
-    return CompletableResultCode.ofSuccess();
+    if (internalTraceExporter.get() == null) {
+      return CompletableResultCode.ofFailure();
+    }
+    return internalTraceExporter.get().flush();
   }
 
   @Override
-  public CompletableResultCode export(Collection<SpanData> spanDataList) {
-    List<Span> spans = new ArrayList<>(spanDataList.size());
-    for (SpanData spanData : spanDataList) {
-      spans.add(translator.generateSpan(spanData, projectId));
+  public CompletableResultCode export(@Nonnull Collection<SpanData> spanDataList) {
+    if (Objects.isNull(internalTraceExporter.get())) {
+      synchronized (this) {
+        try {
+          internalTraceExporter.compareAndSet(null, createActualTraceExporter());
+        } catch (IOException e) {
+          // unable to create actual trace exporter
+          logger.log(
+              Level.WARNING,
+              String.format("Unable to initialize trace exporter. Error %s", e.getMessage()));
+          logger.log(Level.INFO, "Setting TraceExporter to noop.");
+          internalTraceExporter.set(InternalTraceExporter.noop());
```

Comment:
```suggestion
          internalTraceExporter.set(new NoopSpanExporter());
```

IMO, there's no real benefit to having the no-op exporter be a singleton. This would let us get rid of the singleton code as well as the indirection of `SpanExporter.noop()`.

Reply:
My major motivation to do this was the fact that all objects would be exactly same for noop implementation, so in case of multiple TraceExporters, this could be helpful. 

But I can see the `new` call here would be quite inexpensive so I don't have a strong opinion. I can remove singleton implementation here.